internal/storage/storage_map.go

```diff
@@ -1,30 +1,89 @@
 package storage
 
 import (
+	"context"
 	"sync"
+
+	"github.com/google/uuid"
 )
 
 type StorageMap struct {
 	mu   sync.Mutex
-	data map[string]string
+	data map[string]table
 }
 
-func (urls *StorageMap) SaveLongURL(longURL string) (shortURL string, err error) {
+func (storage *StorageMap) SaveLongURL(_ctx context.Context, longURL, userID string) (shortURL string, err error) {
```

Comment:
```suggestion
func (storage *StorageMap) SaveLongURL(_ context.Context, longURL, userID string) (shortURL string, err error) {
```

То же, что и для файлового стораджа.